mergin/client.py

```diff
@@ -915,6 +925,58 @@ def pull_project(self, directory):
         pull_project_wait(job)
         return pull_project_finalize(job)
 
+    def sync_project(self, project_dir):
+        """
+        Syncs project by loop with these steps:
+        1. Pull server version
+        2. Get local changes
+        3. Push first change batch
+        Repeat if there are more changes pending.
```

Comment:
it would be useful to explain a bit what is a "change batch" and what's the philosophy of behind doing multiple batches